.github/workflows/test_suite_ubuntu.yml

```
@@ -214,55 +214,33 @@ jobs:
           . ftorch/bin/activate
           pip install torch torchvision --index-url https://download.pytorch.org/whl/cpu
 
-      # NOTE: Based on github action logs, this installs ifx/ifort, 
-      # icx/icc, and icpx/icpc for v2023.2.0
-      # see https://www.intel.com/content/www/us/en/developer/articles/tool/compilers-redistributable-libraries-by-version.html 
-      - name: Install Intel classic and modern compilers
-        uses: fortran-lang/setup-fortran@d2ba6ea44297a24407def2f6e117954d844a5368 # v1
-        id: setup-fortran
-        with:
-          compiler: intel-classic 
-          version: '2021.10'
-
-      - name: Cache OpenMPI
-        id: cache-openmpi
-        uses: actions/cache@v4
+      - name: Install Intel tools
+        uses: rscohn2/setup-oneapi@8affb2b9c36a17ffc190a96fd27ba5ad9b4b5447 # v0
         with:
-          path: /home/runner/openmpi-${{ matrix.toolchain }}
-          key: openmpi-${{ runner.os }}-openmpi-4.1.2-${{ matrix.toolchain }}
-
-      - name: Build OpenMPI with Intel compilers 
-        if: steps.cache-openmpi.outputs.cache-hit != 'true'
-        run: |
-          wget https://download.open-mpi.org/release/open-mpi/v4.1/openmpi-4.1.2.tar.gz
-          tar -xzf openmpi-4.1.2.tar.gz
-          cd openmpi-4.1.2
-          ./configure \
-            CC=${{ matrix.CC }} CXX=${{ matrix.CXX }} FC=${{ matrix.FC }} \
-            --prefix=$HOME/openmpi-${{ matrix.toolchain }}
-          make -j 8 
-          make install
+          components: |
+            icx@2023.2.1
+            ifx@2023.2.1
+            impi@2021.14.0
 
-      - name: Add OpenMPI to env 
+      - name: Setup Intel oneAPI environment
         run: |
-          echo "PATH=$HOME/openmpi-${{ matrix.toolchain }}/bin:${PATH}" >> $GITHUB_ENV 
-          echo "LD_LIBRARY_PATH=$HOME/openmpi-${{ matrix.toolchain }}/lib:${LD_LIBRARY_PATH}" >> $GITHUB_ENV 
+          source /opt/intel/oneapi/setvars.sh
+          printenv >> $GITHUB_ENV
 
       - name: Install pFUnit
         run: |
-          export FC=${{ matrix.FC }}
-          export CC=${{ matrix.CC }}
-          export CXX=${{ matrix.CXX }}
-          export MPIF90=mpif90
-          export OPENMPI_DIR=$HOME/openmpi-${{ matrix.toolchain }}
           # TODO: Avoid version pinning (needed because version appears in install path)
           git clone -b v4.12.0 https://github.com/Goddard-Fortran-Ecosystem/pFUnit.git
           mkdir pFUnit/build
           cd pFUnit/build
-          cmake .. -DCMAKE_PREFIX_PATH=${OPENMPI_DIR}
+          cmake .. \
+            -DCMAKE_C_COMPILER=${{ matrix.CC }} \
+            -DCMAKE_CXX_COMPILER=${{ matrix.CXX }} \
```

Could these be explicitly specified - to match the changes made to the gnu workflow - and remove them from the matrix leaving just the fortran compiler (which is all we are varying).
```suggestion
            -DCMAKE_C_COMPILER=icx \
            -DCMAKE_CXX_COMPILER=icpx \
```

The same could be said for the standard.

I can also see the argument of leaving them in the matrix so that the full configuration can be seen in one place, however.